src/duckling/game/drawable/tile-block-drawable-drawer.ts

```diff
@@ -15,30 +15,28 @@ export function drawTileBlockDrawable(tileBlockDrawable : TileBlockDrawable, ass
         return null;
     }
     
-    let container = new Container();
-    container.addChild(_constructTileBlockSprite(tileBlockDrawable, baseTexture, assetService));
-    return container;
+    // let container = new Container();
+    return _constructTileBlockSprite(tileBlockDrawable, baseTexture, assetService);
+    // return container;
 }
 
 function _constructTileBlockSprite(tileBlockDrawable : TileBlockDrawable, baseTexture : BaseTexture, assetService : AssetService) : DisplayObject {
     let tileSize = getTileWidth(tileBlockDrawable, assetService);
-    let tiles = { x: tileBlockDrawable.size.x / tileSize, y: tileBlockDrawable.size.y / tileSize};
+    if (tileSize <= 0) throw new Error("Invalid tile size specified");
+    let numberOfTiles = { x: tileBlockDrawable.size.x / tileSize, y: tileBlockDrawable.size.y / tileSize};
     
-    let positions = _getPositions(tiles);
-    let calculateOffset: {x: boolean, y: boolean} = _getOffsets(tiles);
+    let positions = _getPositions(numberOfTiles);
+    let calculateOffset: {x: boolean, y: boolean} = _getOffsets(numberOfTiles);
     
-    return _constructTileBlockContainer(positions, calculateOffset, baseTexture, tileSize, tiles);
+    return _constructTileBlockContainer(positions, calculateOffset, baseTexture, tileSize, numberOfTiles);
 }
 
-function _getPositions(tiles: Vector): Vector[] {
+function _getPositions(numberOfTiles: Vector): Vector[] {
     let positions: Vector[] = [];
     
-    let xPositions: number[] = _getAxisPositions(tiles.x);
-    let yPositions: number[] = _getAxisPositions(tiles.y);
-    
     let position: Vector;
-    for (let xPosition of xPositions) {
-        for (let yPosition of yPositions) {
+    for (let xPosition of _getAxisPositions(numberOfTiles.x)) {
+        for (let yPosition of _getAxisPositions(numberOfTiles.y)) {
             position = { x: xPosition, y: yPosition };
             positions.push(position);
```

Comment:
The position variable does not need to exist. You could replace this with positions.push({x : Xpos, y: Yos});